### ISSUE TITLE
Gradient clipping is not visible in our configs/docs

### ISSUE THREAD

**Issue by RdoubleA** [pytorch/torchtune#1993]
Pointed out in #1992. We support gradient clipping in all our recipes but it is exposed in very few configs (only Llama 3.2 Vision configs: https://github.com/pytorch/torchtune/blob/912af640a422d84ce261c100f804490a5c12e8e7/recipes/configs/llama3_2_vision/90B_full.yaml#L64 We should add `clip_grad_norm: null` to all configs to show that this is a configurable option without affecting the default behavior of other configs.
